### ISSUE TITLE
Linux binary releases

### ISSUE THREAD

**Issue by danyf90** [nicklockwood/SwiftFormat#860]
Hello,
SwiftFormat [support Linux](https://github.com/nicklockwood/SwiftFormat/issues/240), but there is no linux binary attached to the GitHub releases, would it be possible to generate also the linux binary when and attach it to the releases?

Right now if you want to use SwiftFormat on Linux you need to:
- download the source code 
- build it

If you need it in Ci, it means that you will waste more time because:
- you need to have a machine with the whole Swift toolchain to be able to build the source code (hence a larger image)
- you need to compile the code and it takes quite some time

Thank you! 🙌

**Comment by nicklockwood** [comment]
@danyf90 I don't have an actual (non-virtualized) Linux machine to build on, so providing this for every release would be quite an additional burden unfortunately.

**Comment by danyf90** [comment]
@nicklockwood should be very easy to do so with a DockerFile like this:
```
FROM swift:5.3-focal

# Install SwiftFormat 0.47.11
RUN TAG_TO_RELEASE=0.47.11 && \
    git clone --no-checkout https://github.com/nicklockwood/SwiftFormat && \
    cd SwiftFormat && \
    git checkout $TAG_TO_RELEASE && \
    swift build --configuration release && \
    SWIFTFORMAT_BIN_PATH=`swift build --configuration release --show-bin-path` && \
    mv $SWIFTFORMAT_BIN_PATH/swiftformat .

# Add code to upload swiftformat binary to GitHub
```

**Comment by danyf90** [comment]
Hi @nicklockwood 👋 
Sorry to push on this, do you think it will be possible to add this for future releases?

It would greatly help to be able to run swift format in CI without the need to have the whole swift toolchain needed to build swiftformat from sources, which results in much bigger docker images (for example [here](https://hub.docker.com/_/swift?tab=tags&page=1&ordering=last_updated&name=5.3-focal) you can see the slim images size goes down from 500MB to 70MB)

I'm happy to help if there is anything I can do.

Thanks 🙏

**Comment by inket** [comment]
Hi there! Adding a Linux binary would be very useful for us too!

The binary can be built automatically using GitHub Actions so setting this up shouldn't take more than an hour 🙏

**Comment by inket** [comment]
Here's an example of a GitHub Action that builds SwiftFormat on Linux and commits the resulting binary to master.
https://github.com/inket/SwiftFormat/pull/1/files

**Comment by danyf90** [comment]
@nicklockwood have you had the time to check @inket proposal? supporting linux binaries would greatly simplify and speed-up swiftformat usage in CI 🙏

**Comment by nicklockwood** [comment]
I'm not sure that I understand the workflow. When does this action run? If it runs after a tagged release is committed, how can I include the linux binary in the release itself?

**Comment by inket** [comment]
If I'm understanding correctly, you currently create releases manually after building SwiftFormat locally?

If so, I think a manually-triggered workflow would work best. Once the workflow is finished you can download the binary from GitHub and include it in your release 👍 

I'll have a pull request open in a bit.

**Comment by nicklockwood** [comment]
@danyf90 @inket I've included a linux binary with the 0.48.1 release.

**Comment by doozMen** [comment]
I think there is still a minor issue with the now included linux build. It is not used in the artifact bundle. I made a fork and made a new artifactbundle. It basically requires the linux exectuable to be also mentionned. 

https://github.com/doozMen/SwiftFormat/releases/tag/0.49.18 

Could this be included in next release too?
